test/routers/handlers/personal-code/personalCode.unit.ts

```diff
@@ -0,0 +1,67 @@
+import middlewareMocks from "./../../../mocks/allMiddleware.mock";
+import { HttpStatusCode } from "axios";
+import * as cheerio from "cheerio";
+import request from "supertest";
+import app from "../../../../src/app";
+import { PrefixedUrls } from "../../../../src/constants";
+import { COMPANY_NUMBER, CREATED_RESOURCE, PATCH_INDIVIDUAL_STATEMENT_DATA, PSC_VERIFICATION_ID, TRANSACTION_ID } from "../../../mocks/pscVerification.mock";
+import { getUrlWithTransactionIdAndSubmissionId } from "../../../../src/utils/url";
+import { PSC_INDIVIDUAL } from "../../../mocks/psc.mock";
+
+jest.mock("../../../../src/services/pscVerificationService", () => ({
+    getPscVerification: () => ({
+        httpStatusCode: HttpStatusCode.Ok,
+        resource: CREATED_RESOURCE
+    }),
+    patchPscVerification: () => ({
+        httpStatusCode: HttpStatusCode.Ok,
+        resource: PATCH_INDIVIDUAL_STATEMENT_DATA
+    })
+}));
+
+jest.mock("../../../../src/services/pscService", () => ({
+    getPscIndividual: () => ({
+        httpStatusCode: HttpStatusCode.Ok,
+        resource: PSC_INDIVIDUAL
+    })
+}));
+
+beforeEach(() => {
+    jest.clearAllMocks();
+});
+
+describe("personal code router tests", () => {
+
+    beforeEach(() => {
+        middlewareMocks.mockSessionMiddleware.mockClear();
+    });
+
+    afterEach(() => {
+        expect(middlewareMocks.mockSessionMiddleware).toHaveBeenCalledTimes(1);
+    });
+
+    it("Should render the Personal Code page with a success status code and correct links", async () => {
+        const queryParams = new URLSearchParams("lang=en");
+        const uriWithQuery = `${PrefixedUrls.PERSONAL_CODE}?${queryParams}`;
+        const uri = getUrlWithTransactionIdAndSubmissionId(uriWithQuery, TRANSACTION_ID, PSC_VERIFICATION_ID);
+
+        const resp = await request(app).get(uri);
+
+        const $ = cheerio.load(resp.text);
+
+        expect(resp.status).toBe(HttpStatusCode.Ok);
+        expect($("a.govuk-back-link").attr("href")).toBe("/persons-with-significant-control-verification/transaction/11111-22222-33333/submission/662a0de6a2c6f9aead0f32ab/individual/psc-list?lang=en");
+    });
+
+    it("Should redirect to Individual Statement screen", async () => {
```

Comment:
ah I wonder if this test should be in the router test?  I'm a bit unsure of what type of tests should go where - router / handler, int / unit.

Reply:
same as above - following dev discussion 11/6 we will wait for the routers and handlers to be refactored, and part of that will be re-organising the tests.  So my comment can be covered there.